backend/src/main/java/com/example/backend/controller/GoalsController.java

```diff
@@ -43,4 +43,29 @@ public ResponseEntity<?> createGoalElement(@RequestParam String goalName, @Reque
         GoalElement savedElement = goalElementRepository.save(element);
         return ResponseEntity.created(new URI("/goals/elements/" + savedElement.getId())).body(savedElement);
     }
+
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity deleteGoal(@PathVariable Long id){
+        User user = userRepository.findTopByOrderByIdAsc();
+        Goal goal = goalRepository.getById(id);
+        goalRepository.deleteById(id);
+        List<GoalElement> to_delete = goalElementRepository.findGoalElementsByGoal(goal);
+        for(GoalElement subgoal: to_delete){
+            goalElementRepository.deleteById(subgoal.getId());
+        }
+        return ResponseEntity.ok().build();
+    }
+
```

Comment:
Jako wartość zwracaną możesz dać ResponseEntity<?> to wtedy kompilator się nie będzie rzucał, że "Raw use of parameterized class 'ResponseEntity' ", ale to drobnostka.
Dodatkowo zmienną user w deleteGoal można usunąć, bo z niej nigdy nie korzystasz

Reply:
Sama to powinnam zmienić w IncomesExpensesController, ale odkryłam to po fakcie xd